edb/schema/links.py

```diff
@@ -40,14 +40,20 @@
 from . import utils
 
 if TYPE_CHECKING:
+    from . import objtypes as s_objtypes
     from . import schema as s_schema
 
 
 LinkTargetDeleteAction = qltypes.LinkTargetDeleteAction
 
 
-def merge_actions(target: so.Object, sources: List[so.Object],
-                  field_name: str, *, schema) -> object:
+def merge_actions(
+    target: so.InheritingObject,
+    sources: List[so.Object],
+    field_name: str,
+    *,
+    schema: s_schema.Schema,
+) -> object:
```

Comment:
I know you didn't change this now but I noticed this. Returning `object` from a function means that you declare that the only safe methods to call and attributes to access are those that are present on the base `object` type, which is very little. Such a returned value is very annoying to use.

Did you mean `Any` here?

Reply:
Thanks, Łukasz. I just looked into this, and indeed `Any` is the right annotation here: the function `merge_actions` is used as `merge_fn`, whose protocol is defined in `schema.objects` as returning Any. Pushing to fix this.